pytype/file_utils.py

```diff
@@ -136,3 +140,12 @@ def replace_separator(path: str):
     return path.replace("/", os.path.sep).replace(":", os.pathsep)
   else:
     return path
+  
+def is_file_script(filename, dir=None):
+  # This is for python files that do not have the .py extension
+  # of course we assume that they start with a shebang
+  file_path = expand_path(filename, dir)
+  if path_utils.isfile(file_path):
+    with open(file_path, 'r') as file:
+      line = file.readline().rstrip().lower()
+      return line.startswith('#!') and (line.endswith('python') or line.endswith('python3'))
```

Comment:
I'd recommend using a regex here. Something like `return re.fullmatch(r'#!.+python3?', line) is not None` should work.